src/app/body/all-products/filters/filters.component.ts

```diff
@@ -1,15 +1,138 @@
-import { Component, OnInit } from '@angular/core';
+import { Component, ElementRef, OnInit } from '@angular/core';
+import { ProductService } from '../../../services/product.service';
+import { Product } from '../../../models/product.model';
+import { FormArray, FormControl, FormGroup } from '@angular/forms';
+import { Options } from '@angular-slider/ngx-slider';
 
 @Component({
   selector: 'app-filters',
   templateUrl: './filters.component.html',
   styleUrls: ['./filters.component.scss']
 })
 export class FiltersComponent implements OnInit {
+  private products: Product[];
+  public form: FormGroup;
+  public categories: string[];
+  public brands: string[];
+  public rating = Array(5);
+  public sliderOptions: Options;
+  public selectedValue = 'Select';
+  public sortOptions = [
+    {value: 'price', operator: '-', label: 'Price: Low to High'},
```

Comment:
This should be in separated file

Reply:
fixed this and 'eval' func with sort method